### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by singularo**
Sorry, re-doing PR from this repo so you can just set the branch to dev-develop in composer.json to test.

Why
Gives a closer to production env for developers
Solves all known permissions issues by running apache2 as the uid of the developer
Slight pain in that it does a build of the image locally to decorate it

Details
Add Dockerfile that is used to decorate the public/production s2i image for development
Adjustments to dsh and docker-compose for linux and osx to support updated image
Cache prefix support added to settings to allow intermediate redis mitigation on Uni hosting.
Infinispan config file added and update process for all files adjusted for new files

Testing requires local build of https://github.com/universityofadelaide/s2i-shepherd-drupal/pull/15, updating composer.json and removing dsh & docker-compose.*.yml so that they get updated.

### CODE REVIEW

File: standalone-memcached.xml [hunk 1]
```diff
@@ -0,0 +1,351 @@
+<?xml version='1.0' encoding='UTF-8'?>
```
Comment (acbramley) [inline standalone-memcached.xml]: Think this file should go into a config directory or something?

File: src/Handler.php [hunk 2]
```diff
@@ -171,12 +173,22 @@ public function generateSettings()
...
             "  \$settings['cache']['bins']['discovery'] = 'cache.backend.chainedfast';\n" .
             "  \$settings['cache']['bins']['config'] = 'cache.backend.chainedfast';\n\n" .
+            "  \$settings['cache_prefix']['default'] = getenv('REDIS_PREFIX') ?: 'mysite_';\n" .
             "  // If we're not installing, include the redis services.\n" .
             "  if (!isset(\$GLOBALS['install_state'])) {\n" .
             "    \$settings['cache']['default'] = 'cache.backend.redis';\n\n" .
             "    \$settings['container_yamls'][] = 'modules/contrib/redis/example.services.yml';\n" .
             "  }\n" .
             "}\n" .
+            "if (getenv('MEMCACHE_ENABLED')) {\n" .
```
Comment (acbramley) [inline src/Handler.php]: So we're moving from redis to memcache?

Comment (singularo) [reply src/Handler.php]: memcached interface to talk to infinispan (Redhat Data Grid).

File: standalone-memcached.xml (hunk 1 above)
Comment (singularo) [reply standalone-memcached.xml]: Thought about that, but making a new dir for one file seems not worth it. Still have the same number of dir entries, and makes it (slightly) harder to find the config. If it was more files, yes. If we can move some others, yes. Open to suggestions.

Comment (acbramley) [reply standalone-memcached.xml]: You could have a `docker` directory and chuck the Dockerfile in there too?

Comment (singularo) [reply standalone-memcached.xml]: Yep, makes sense. moved.

File: docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+RUN wget -q https://getcomposer.org/installer -O - | php -d allow_url_fopen=On -- --install-dir=/usr/local/bin --filename=composer
+
+RUN echo "zend_extension=xdebug.so" > /etc/php/7.2/mods-available/xdebug.ini \
+&& echo "xdebug.max_nesting_level=500" >> /etc/php/7.2/mods-available/xdebug.ini \
+&& echo "xdebug.remote_enable=1" >> /etc/php/7.2/mods-available/xdebug.ini \
+&& echo "xdebug.remote_handler=dbgp" >> /etc/php/7.2/mods-available/xdebug.ini \
+&& echo "xdebug.remote_mode=req" >> /etc/php/7.2/mods-available/xdebug.ini \
+&& echo "xdebug.remote_port=9000" >> /etc/php/7.2/mods-available/xdebug.ini \
+&& echo "xdebug.remote_connect_back=1" >> /etc/php/7.2/mods-available/xdebug.ini \
+&& echo "xdebug.idekey=PHPSTORM" >> /etc/php/7.2/mods-available/xdebug.ini
```
Comment (acbramley) [inline docker/Dockerfile]: Could just have a xdebug.ini in the repo and COPY that into the container?

**Pull request merged**